GVFS/GVFS.Common/NamedPipes/LockNamedPipeMessages.cs

```diff
@@ -317,23 +320,46 @@ internal static LockData FromBody(string body)
 
                     // ParsedCommandLength should be the length of the string at the end of the message
                     // Add the length of the previous parts, plus delimiters
-                    int startingSpot = dataParts[0].Length + dataParts[1].Length + dataParts[2].Length + dataParts[3].Length + 4;
-                    if ((startingSpot + parsedCommandLength) != body.Length)
+                    int commandStartingSpot = dataParts[0].Length + dataParts[1].Length + dataParts[2].Length + dataParts[3].Length + 4;
+                    if ((commandStartingSpot + parsedCommandLength) >= body.Length)
                     {
                         throw new InvalidOperationException(string.Format("Invalid lock message. The parsedCommand is an unexpected length, got: {0} from message: '{1}'", parsedCommandLength, body));
                     }
 
-                    parsedCommand = body.Substring(startingSpot, parsedCommandLength);
+                    parsedCommand = body.Substring(commandStartingSpot, parsedCommandLength);
+
+                    string sessionIdSubString = body.Substring(commandStartingSpot + parsedCommandLength + 1);

```

Comment:
Need to add some comments about these calculations.  This it ugly and I wish we were simply using `JSON` for the message data especially when both sides are `C#`.  Not sure of the performance impact though.